packages/notifications/__tests__/testUtils/mergeInAppMessageWithOverrides.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { cloneDeep } from "lodash";
+import { InAppMessageCampaign, OverrideButtonConfiguration } from "@aws-amplify/core/internals/aws-clients/pinpoint";
+import { InAppMessageButton, InAppMessageContent } from "../../src/inAppMessaging/types/message";
+import { mapOSPlatform } from "../../src/inAppMessaging/providers/pinpoint/utils/helpers";
+
+export const mergeInAppMessageWithOverrides:(
+	pinpointInAppMessage:InAppMessageCampaign,
+	platform:'android'|'ios'|'web',
+	buttonOverrides?:{
+		primaryButton:OverrideButtonConfiguration,
+		secondaryButton:OverrideButtonConfiguration
+	}
+ ) =>InAppMessageCampaign = (pinpointInAppMessage, platform, buttonOverrides)=>{
+	const message= cloneDeep(pinpointInAppMessage);
+	const configPlatform = mapOSPlatform(platform);
+	if (message?.InAppMessage?.Content) {
+		message.InAppMessage.Content[0] = {
+			...message.InAppMessage.Content[0],
+			PrimaryBtn:{
+				...message.InAppMessage.Content[0].PrimaryBtn,
+				[configPlatform]:buttonOverrides?.primaryButton
+			},
+			SecondaryBtn:{
+				...message.InAppMessage.Content[0].SecondaryBtn,
+				[configPlatform]:buttonOverrides?.secondaryButton
+			}
+		} 
+	}
+	return message
+}
+
+export const mergeExpectedContentWithExpectedOverride:(
+	inAppMessage:InAppMessageContent,
+	expectedButtonConfig: {
+		primaryButton:OverrideButtonConfiguration,
+		secondaryButton:OverrideButtonConfiguration
+	}
+)=>InAppMessageContent = (inAppMessage, expectedButtonConfig)=>{
```

Comment:
Nit: Ditto here, could tidy up the typing a bit
```suggestion
export const mergeExpectedContentWithExpectedOverride = (
	inAppMessage: InAppMessageContent,
	expectedButtonConfig: {
		primaryButton: OverrideButtonConfiguration;
		secondaryButton: OverrideButtonConfiguration;
	}
): InAppMessageContent => {
```

Reply:
updated here:
[updated merge override fns signature and associated tests](https://github.com/aws-amplify/amplify-js/pull/12917/commits/3a58f157071dcb7ee62a9e8bf062c96bb24d7cd0)